aiida/cmdline/commands/cmd_database.py

```diff
@@ -8,95 +8,60 @@
+)
 def database_version():
     """Show the version of the database.
 
     The database version is defined by the tuple of the schema generation and schema revision.
-    """
-    from aiida.manage.manager import get_manager
-
-    manager = get_manager()
-    manager._load_backend(schema_check=False)  # pylint: disable=protected-access
-    backend_manager = manager.get_backend_manager()
 
-    echo.echo('Generation: ', bold=True, nl=False)
-    echo.echo(backend_manager.get_schema_generation_database())
-    echo.echo('Revision:   ', bold=True, nl=False)
-    echo.echo(backend_manager.get_schema_version_backend())
+    .. deprecated:: v2.0.0
+    """
 
 
 @verdi_database.command('migrate')
 @options.FORCE()
-def database_migrate(force):
-    """Migrate the database to the latest schema version."""
-    from aiida.engine.daemon.client import get_daemon_client
-    from aiida.manage.manager import get_manager
-
-    client = get_daemon_client()
-    if client.is_daemon_running:
-        echo.echo_critical('Migration aborted, the daemon for the profile is still running.')
-
-    manager = get_manager()
-    profile = manager.get_profile()
-    backend = manager._load_backend(schema_check=False)  # pylint: disable=protected-access
-
-    if force:
-        try:
-            backend.migrate()
-        except (exceptions.ConfigurationError, exceptions.DatabaseMigrationError) as exception:
-            echo.echo_critical(str(exception))
-        return
-
-    echo.echo_warning('Migrating your database might take a while and is not reversible.')
-    echo.echo_warning('Before continuing, make sure you have completed the following steps:')
-    echo.echo_warning('')
-    echo.echo_warning(' 1. Make sure you have no active calculations and workflows.')
-    echo.echo_warning(' 2. If you do, revert the code to the previous version and finish running them first.')
-    echo.echo_warning(' 3. Stop the daemon using `verdi daemon stop`')
-    echo.echo_warning(' 4. Make a backup of your database and repository')
-    echo.echo_warning('')
-    echo.echo_warning('', nl=False)
-
-    expected_answer = 'MIGRATE NOW'
-    confirm_message = 'If you have completed the steps above and want to migrate profile "{}", type {}'.format(
-        profile.name, expected_answer
-    )
-
-    try:
-        response = click.prompt(confirm_message)
-        while response != expected_answer:
-            response = click.prompt(confirm_message)
-    except click.Abort:
-        echo.echo('\n')
-        echo.echo_critical('Migration aborted, the data has not been affected.')
-    else:
-        try:
-            backend.migrate()
-        except (exceptions.ConfigurationError, exceptions.DatabaseMigrationError) as exception:
-            echo.echo_critical(str(exception))
-        else:
-            echo.echo_success('migration completed')
+@click.pass_context
+@decorators.deprecated_command(
+    'This command has been deprecated and will be removed soon (in v3.0). '
```

Comment:
Is it intentional that this will be removed in 3.0 and the rest in 2.1?

Reply:
Yes, this is intentional. I personally would prefer to leave the deprecations for longer (not sure if until 3.0, but probably more than 2.1 since I suspect this will be out relatively soon), but this is what was agreed in the last meeting (see [notes](https://docs.google.com/document/d/1WpC-eFK4jXpb581Pfmxv2tLrMMZT9XLg2hH4Z8rQDOo/edit#heading=h.dtdzkrsqjy)).

BTW: I'm still not convinced about exposing this information to the user since this is mostly for internal reference (i.e. remembering when to take it out), but I didn't know where else to track this since the `.. deprecated:: vA.B.C` addition to the docstring is for recording when it was deprecated and not up to when. Any suggestions?

Reply:
I don't agree that it is for internal remembering. Since it affects the user exactly when it is going to be removed, it seems like very relevant information for them. Only thing missing is _when_ the version that it will be removed in is expected to be released.